lib/vec_deque.h

```diff
@@ -0,0 +1,15 @@
+#pragma once
+
+#include <stddef.h>
+#include <stdint.h>
+
+typedef struct VecDeque VecDeque;
+
+VecDeque* vec_deque_init(size_t size);
+void vec_deque_push_back(VecDeque* deque, void* elem);
+void vec_deque_push_front(VecDeque* deque, void* elem);
+void* vec_deque_pop_back(VecDeque* deque);
+void* vec_deque_pop_front(VecDeque* deque);
+size_t vec_deque_len(VecDeque* deque);
```

Comment:
Nit: I don't care either way, but would be nice to be consistent with our data structure APIs -- so far we've been using `_length` (rather than `_len`).

Reply:
On it! :)